tests/python/test_basic_pathlike.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+import sys
+from contextlib import contextmanager
+try:
+    # python 2
+    from StringIO import StringIO
+except ImportError:
+    # python 3
+    from io import StringIO
+import numpy as np
+import xgboost as xgb
+import unittest
+import json
+import pytest
+import testing as tm
+
+# This is the select basic tests from test_basic.py with modifications for os.PathLike
+# instead of string file paths throughout.
+dpath = 'demo/data/'
+
+try:
+    from pathlib import Path
+    dpath = Path(dpath)
+except ImportError:
+    pass
+
+pytestmark = pytest.mark.skipif(**tm.no_pathlike())
+
+rng = np.random.RandomState(1994)
+
+class TestBasicPathLike(unittest.TestCase):
+
+    def test_basic_PathLike(self):
```

Comment:
I don't think it takes a complete run of training to test this.  Try unittest.

Reply:
Will do, I was aiming for consistency with `test_basic` since these are relatively fast, but agreed a full run likely isn't needed. I'll remove this separated file and integrate with the other tests directly.